cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,182 @@
+	updateSeriesClient updateSeriesAPI
+
+	applicationName string
+	force           bool
+	machineNumber   string
+	series          string
+}
+
+var updateSeriesDoc = `
+It is recommended to update the series on existing units before running this
+command to update a machines' series.
+
+By updating the application series, future units of the application to use
+the new series.
+
+The update is disallowed unless the --force flag is used if the requested
+series is not explicitly supported by the application's charm and any
+subordinates, as well as any other charms which may be deployed to the same
+machine.
+
+The --force option should be used with caution since using a charm on a machine
+running an unsupported series may cause unexpected behavior.  Alternately, if
+the requested series supported in later revisions of the charm, upgrade-charm
+can run beforehand.
+
+Updating the
+
+Examples:
+	juju update-series <application> <series>
+	juju update-series <application> <series> --force
+	juju update-series <machine> <series>
+	juju update-series <machine> <series> --force
+
+See also:
+    machines
+    status
+    upgrade-charm
+`
+
+func (c *updateSeriesCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-series",
+		Args:    "[<application>|<machine>] <series>",
+		Purpose: "Update an application or machine's series.",
+		Doc:     updateSeriesDoc,
+	}
+}
+
+func (c *updateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.force, "force", false, "Update even if the series is not supported by the charm and/or related subordinate charms.")
+}
+
+// Init implements cmd.Command.
+func (c *updateSeriesCommand) Init(args []string) error {
+	switch len(args) {
+	case 2:
+		if names.IsValidApplication(args[0]) {
+			c.applicationName = args[0]
+		} else if names.IsValidMachine(args[0]) {
+			c.machineNumber = args[0]
+		} else {
+			return errors.Errorf("invalid application name or machine number %q", args[0])
+		}
+		c.series = args[1]
+	case 1, 0:
+		return errors.Errorf("No application/machine and/or series specified")
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+	return nil
+}
+
+// Run connects to the specified environment and starts the application or machine
+// update process.
+func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
+
+	if c.updateSeriesClient == nil && c.charmClient == nil {
+		apiRoot, err := c.NewAPIRoot()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		c.apiRoot = apiRoot
+		c.updateSeriesClient = application.NewClient(c.apiRoot)
+		c.charmClient = charms.NewClient(c.apiRoot)
+	}
+
+	if c.apiRoot.BestFacadeVersion("Application") < 5 {
+		suffix := "this server"
+		if version, ok := c.apiRoot.ServerVersion(); ok {
+			suffix = fmt.Sprintf("server version %s", version)
+		}
+		return errors.New("updating the application series is not supported by " + suffix)
+	}
+
+	if c.machineNumber != "" {
+		//return c.updateMachineSeries(ctx)
```

Comment:
Add a TODO, and return a not supported error so that it's obvious to the user testing this that update machine series is not supported (yet)